### PR TITLE
[openvino-optimum] Fix BERT accuracy issue and other updates

### PR DESCRIPTION
**Pull request by helena-intel**
- Add token_type_ids for PyTorch models. This fixes an accuracy issue with some BERT models.
- Add accuracy check tests (would catch accuracy issue fixed by adding token_type_ids)
- Get output names for QA models from model instead of hardcoding (required for NNCF models unless users manually set the expected output names)
- Use ONNX opset_version 12
- Add datasets to [nncf] and [all] requirements

token_type_ids not added to TensorFlow models yet - that requires more testing and tests and should be fixed in a future PR. Added a #TODO comment to TF model loading for now.

requirements could be improved further to distinguish between nncf and all. You need datasets for NNCF examples though, and it's a simple library - it is good to have it installed by default with nncf too.

### CODE REVIEW

File: modules/optimum/optimum/intel/openvino/modeling_ov_utils.py [hunk 1]
```diff
@@ -51,20 +48,30 @@ def load_ov_model_from_pytorch(model, inputs=None):
     import io
 
     buf = io.BytesIO()
+
+    tokenizer = AutoTokenizer.from_pretrained(model.name_or_path)
```
Comment (helena-intel) [inline modules/optimum/optimum/intel/openvino/modeling_ov_utils.py:52]: Having to initialize a tokenizer just to get model input names is not ideal but I did not find another reliable way to get these names.

File: modules/optimum/tests/openvino/test_modeling_ov_auto.py [hunk 2]
```diff
@@ -98,7 +98,62 @@ def test_from_ir(self):
...
+    @require_torch
+    def test_tinybert_from_pt(self):
+        model_name = "Intel/dynamic_tinybert"
+        tok = AutoTokenizer.from_pretrained(model_name)
+        model = OVAutoModelForQuestionAnswering.from_pretrained(model_name, from_pt=True)
+        self.check_model(model, tok, "if it makes me grow smaller, i can creep under the door")
+
+
+@require_torch
+class SameOutputTest(unittest.TestCase):
```
Comment (mzegla) [inline modules/optimum/tests/openvino/test_modeling_ov_auto.py:112]: Could we have more descriptive name? `MultiFrameworkSameOutputTest` or something like that?

File: modules/optimum/optimum/intel/openvino/modeling_ov_utils.py [hunk 3]
```diff
@@ -51,20 +48,34 @@ def load_ov_model_from_pytorch(model, inputs=None):
...
+
+    try:
+        tokenizer = AutoTokenizer.from_pretrained(model.name_or_path)
+        input_names = tokenizer.model_input_names
+    except OSError:
+        input_names = [model.main_input_name, "attention_mask"]
+
+    if "token_type_ids" in input_names:
+        # token_type_ids should be last input
+        input_names = [model.main_input_name, "attention_mask", "token_type_ids"]
```
[the remark this answers is not in the document]
Comment (helena-intel) [reply modules/optimum/optimum/intel/openvino/modeling_ov_utils.py:60]: I'm not a fan of hardcoding the input names like this, but this line is necessary to get token_type_ids as last input. Tests failed when I did not do that. Currently input_names are set the same way:
```
        input_names = [model.main_input_name, "attention_mask"]
```

File: modules/optimum/optimum/intel/openvino/modeling_ov_utils.py [hunk 4]
```diff
@@ -51,20 +48,34 @@ def load_ov_model_from_pytorch(model, inputs=None):
...
         else:
-            inputs = (dummy_input_ids, dummy_mask)
-
-        input_names = [model.main_input_name, "attention_mask"]
+            inputs = tuple(
+                [
+                    dummy_inputs,
+                ]
+                * len(input_names)
+            )
```
[the remark this answers is not in the document]
Comment (helena-intel) [reply modules/optimum/optimum/intel/openvino/modeling_ov_utils.py:78]: This is done by black. I much prefer the shorter format in this case too, but there's a CI check that checks that black formatting is used. It is also nice not having to think about formatting and just let black handle that. But if you want, I can look into adding nofmt tags before and after this line to tell black to ignore it.

File: modules/optimum/tests/openvino/test_modeling_ov_auto.py (hunk 2 above)
Comment (helena-intel) [reply modules/optimum/tests/openvino/test_modeling_ov_auto.py:112]: Yes, I'll change that, thanks!

File: modules/optimum/optimum/intel/openvino/modeling_ov_utils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mzegla) [reply modules/optimum/optimum/intel/openvino/modeling_ov_utils.py:78]: Oh I see. I agree that it's nice to rely on formatting tools and linters. I'm okay with it in that case.

**Review by mzegla — APPROVED** [review]
